### PR TITLE
Permit primary key field changes on update

### PR DESCRIPTION
**Pull request by jhawkwind**
This will allow primary keys to be changed. It is up to the end
developer to handle any integrity issues, security risks, or foreign key
relation issues as a result of updating the primary key column.

This will resolve issue https://github.com/cakephp/cakephp/issues/6465
with this ORM enhancement.

### CODE REVIEW

File: tests/TestCase/ORM/TableTest.php [hunk 1]
```diff
@@ -2394,36 +2410,94 @@ public function testSaveUpdatePrimaryKeyNotModified()
...
+            'password' => '$2a$10$u05j8FjsvLBNdfhBhc21LOuVMpzpabVXQ9OpC2wO3pSO0q6t7HHMO',
+            'created' => '2012-06-10 01:22:23', 'updated' => '2015-05-21 15:43:00'];
+        
+        $entity = $this->getMock(
+            '\Cake\ORM\Entity',
+            null,
+            [$originalData, ['markNew' => false]]
+        );
+        
+        $this->assertSame($originalData, $entity->toArray());
```
Comment (markstory) [inline tests/TestCase/ORM/TableTest.php]: Why are you using a mock here? It seems unnecessary

Comment (jhawkwind) [inline tests/TestCase/ORM/TableTest.php]: Well, originally the old test was doing so, would you prefer that I modify existing data in the Fixture?

Comment (markstory) [inline tests/TestCase/ORM/TableTest.php]: I don't think you need a fixture, you can create instances of Entity() using the constructor.

Comment (jhawkwind) [inline tests/TestCase/ORM/TableTest.php]: OH! I am sorry, I misunderstood you as to which part needing a getMock. That is a good question, do not know how I ended up with it. Going to correct that.

**Comment by jhawkwind** [comment]
So, according to AppVeyor and some research, updating primary key on MSSQL will fail (while all other flavors it will succeed and the FKs will be updated if set that way). How should this be handled? Both from a test stance and a the programming end?

**Comment by markstory** [comment]
In other situations where SQLserver is the odd one out we've split the tests up and skipped the failing parts when the current driver is sqlserver.

**Comment by markstory** [comment]
Still lots of failing tests on SQLServer. We'll need to get those passing before this can be merged.

**Comment by jhawkwind** [comment]
Sorry, I am going to close this, and resubmit a new pull request when I get the opportunity to follow up on this.

**Comment by markstory** [comment]
Ok.

**Pull request closed without merging**